notebooks/approx-nearest-neighbors.ipynb

```diff
@@ -0,0 +1,466 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "db673eda-86ee-47a7-975d-aa3d36c2f407",
+   "metadata": {},
+   "source": [
+    "# Approximate Nearest Neighbors"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eeb78fae-ae08-4b64-8daa-b579af1d9ba6",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "from spark_rapids_ml.knn import ApproximateNearestNeighbors\n",
+    "from pyspark.sql.functions import col"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "17955bd7-b911-4da6-ad1b-02ae4a5f0a2b",
+   "metadata": {},
+   "source": [
+    "### Create synthetic dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38c671ba-b6de-4414-992f-957e28f3a8be",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "dim = 2000\n",
+    "dtype = 'float32'\n",
+    "np.random.seed(1)\n",
+    "\n",
+    "# items\n",
+    "num_vecs = 100000\n",
+    "vec = np.random.randn(dim).reshape([1,dim])\n",
+    "arr = np.random.randn(num_vecs).reshape([num_vecs,1])\n",
+    "items = arr * vec\n",
```

Comment:
We might want to have a 'better' selection of vectors.  I see this is also in the exact knn notebook but didn't notice before.   The vectors are all multiples of a single vectors.   Perhaps we can find a better choice of vectors for the next release.